### PR TITLE
Introducing granularity enum

### PR DESCRIPTION
**Pull request by edrickrenan**
## Description

We now have a centralized place for all the granularity supported by the bot for the different exchanges.
In this first step, I changed the Binance class to use the Enum, and if this PR is merged, I'll add it to the other exchanges.

Fixes # (issue)

## Type of change

Please make your selection.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change requires a new dependency
- [x] Code Maintainability / comments
- [x] Code Refactoring / future-proofing

## How Has This Been Tested?

I tested it locally with my bot, both running the smart switch and granularity coming from the config.json

I may need some help from people using other exchanges just to confirm that my changes haven't broken the bot for them.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added pytest unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pycryptobot.py [hunk 1]
```diff
@@ -1674,12 +1675,13 @@ def executeJob(
 
 def main():
     try:
+        _websocket = None
```
Comment (edrickrenan) [inline pycryptobot.py:1678]: I'll revert this change.

**Comment by edrickrenan** [comment]
Running the bot for Binance in sim mode:
```
--------------------------------------------------------------------------------
|                 *** Getting smartswitch (1h) market data ***                 |
================================================================================
2021-10-23 04:00:00 | ARBUSD | 1h | Close: 54.31 | v EMA12/26: 54.4073 < 54.5314 v | v MACD: -0.1242 < -0.1222 v | WAIT | Last Action: SELL | DF HIGH: 63.47 | DF LOW: 49.34 | SWING: 28.64% | CURR Price is -14.43% away from DF HIGH | Range: 2021-10-10 17:00:00 <--> 2021-10-23 04:00:00
2021-10-23 05:00:00 | ARBUSD | 1h | Close: 53.94 | v EMA12/26: 54.3354 < 54.4876 v | v MACD: -0.1522 < -0.1282 v | WAIT | Last Action: SELL | DF HIGH: 63.47 | DF LOW: 49.34 | SWING: 28.64% | CURR Price is -15.01% away from DF HIGH | Range: 2021-10-10 18:00:00 <--> 2021-10-23 05:00:00
2021-10-23 06:00:00 | ARBUSD | 1h | Close: 53.7 | v EMA12/26: 54.2377 < 54.4293 v | v MACD: -0.1917 < -0.1409 v | WAIT | Last Action: SELL | DF HIGH: 63.47 | DF LOW: 49.34 | SWING: 28.64% | CURR Price is -15.39% away from DF HIGH | Range: 2021-10-10 19:00:00 <--> 2021-10-23 06:00:00
* Candlestick Detected: Hammer ("Weak - Reversal - Bullish Signal - Up")
2021-10-23 07:00:00 | ARBUSD | 1h | Close: 53.75 | v EMA12/26: 54.1626 < 54.3790 v | v MACD: -0.2164 < -0.1560 v | WAIT | Last Action: SELL | DF HIGH: 63.47 | DF LOW: 49.34 | SWING: 28.64% | CURR Price is -15.31% away from DF HIGH | Range: 2021-10-10 20:00:00 <--> 2021-10-23 07:00:00
*** smart switch from granularity 3600 (1 hour) to 900 (15 min) ***
2021-10-23 08:00:00 | ARBUSD | 15m | Close: 53.87 | v EMA12/26: 54.1176 < 54.3413 v | v MACD: -0.2237 < -0.1695 v | WAIT | Last Action: SELL | DF HIGH: 63.47 | DF LOW: 49.34 | SWING: 28.64% | CURR Price is -15.13% away from DF HIGH | Range: 2021-10-10 21:00:00 <--> 2021-10-23 08:00:00
--------------------------------------------------------------------------------
|                *** Getting smartswitch (15m) market data ***                 |
================================================================================
2021-10-23 08:15:00 | ARBUSD | 15m | Close: 53.78 | v EMA12/26: 53.7917 < 53.9645 v | v MACD: -0.1729 < -0.1651 v | WAIT | Last Action: SELL | DF HIGH: 58.03 | DF LOW: 52.96 | SWING: 9.57% | CURR Price is -7.32% away from DF HIGH | Range: 2021-10-20 05:30:00 <--> 2021-10-23 08:15:00
```

Running the bot in sim mode for coinbasepro:
```
--------------------------------------------------------------------------------
|                 *** Getting smartswitch (1h) market data ***                 |
================================================================================
2021-10-23 04:00:00 | BTC-USD | 3600 | Close: 61152.89 | v EMA12/26: 61159.7183 < 61703.9329 v | ^ MACD: -544.2147 > -643.5149 ^ | WAIT | Last Action: SELL | DF HIGH: 66890.92 | DF LOW: 54608.07 | SWING: 22.49% | CURR Price is -8.58% away from DF HIGH | Range: 2021-10-10 17:00:00 <--> 2021-10-23 04:00:00
2021-10-23 05:00:00 | BTC-USD | 3600 | Close: 60983.66 | v EMA12/26: 61132.6324 < 61650.5794 v | ^ MACD: -517.9470 > -618.4013 ^ | WAIT | Last Action: SELL | DF HIGH: 66890.92 | DF LOW: 54608.07 | SWING: 22.49% | CURR Price is -8.83% away from DF HIGH | Range: 2021-10-10 18:00:00 <--> 2021-10-23 05:00:00
* Candlestick Detected: Hammer ("Weak - Reversal - Bullish Signal - Up")
2021-10-23 06:00:00 | BTC-USD | 3600 | Close: 61041.62 | v EMA12/26: 61118.6305 < 61605.4713 v | ^ MACD: -486.8408 > -592.0892 ^ | WAIT | Last Action: SELL | DF HIGH: 66890.92 | DF LOW: 54608.07 | SWING: 22.49% | CURR Price is -8.74% away from DF HIGH | Range: 2021-10-10 19:00:00 <--> 2021-10-23 06:00:00
2021-10-23 07:00:00 | BTC-USD | 3600 | Close: 60868.89 | v EMA12/26: 61080.2089 < 61550.9097 v | ^ MACD: -470.7009 > -567.8116 ^ | WAIT | Last Action: SELL | DF HIGH: 66890.92 | DF LOW: 54608.07 | SWING: 22.49% | CURR Price is -9.0% away from DF HIGH | Range: 2021-10-10 20:00:00 <--> 2021-10-23 07:00:00
*** smart switch from granularity 3600 (1 hour) to 900 (15 min) ***
2021-10-23 08:00:00 | BTC-USD | 900 | Close: 61075.3 | v EMA12/26: 61079.4537 < 61515.6794 v | ^ MACD: -436.2257 > -541.4944 ^ | WAIT | Last Action: SELL | DF HIGH: 66890.92 | DF LOW: 54608.07 | SWING: 22.49% | CURR Price is -8.69% away from DF HIGH | Range: 2021-10-10 21:00:00 <--> 2021-10-23 08:00:00
--------------------------------------------------------------------------------
|                *** Getting smartswitch (15m) market data ***                 |
================================================================================
2021-10-23 08:15:00 | BTC-USD | 900 | Close: 61050.22 | v EMA12/26: 60979.1957 < 61001.4389 v | v MACD: -22.2433 < -14.1254 v | WAIT | Last Action: SELL | DF HIGH: 66890.92 | DF LOW: 60189.99 | SWING: 11.13% | CURR Price is -8.73% away from DF HIGH | Range: 2021-10-20 05:30:00 <--> 2021-10-23 08:15:00
2021-10-23 08:30:00 | BTC-USD | 900 | Close: 60948.74 | v EMA12/26: 60974.5102 < 60997.5353 v | v MACD: -23.0252 < -15.9054 v | WAIT | Last Action: SELL | DF HIGH: 66890.92 | DF LOW: 60189.99 | SWING: 11.13% | CURR Price is -8.88% away from DF HIGH | Range: 2021-10-20 05:45:00 <--> 2021-10-23 08:30:00
```

Granularity is still working as expected

**Comment by whittlem** [comment]
Thanks @edrickrenan. There is a conflict in the models/PyCryptoBot.py. Can you review and resolve please? Thanks

**Comment by edrickrenan** [comment]
> Thanks @edrickrenan. There is a conflict in the models/PyCryptoBot.py. Can you review and resolve please? Thanks

Done, @whittlem!

**Pull request merged**